src/wristpy/processing/analytics.py

```diff
@@ -0,0 +1,330 @@
+            max_wakeup = None
+            for inactivity_bout in sib_periods:
+                if (
+                    inactivity_bout[0] <= sleep_guide[1]
+                    and inactivity_bout[1] >= sleep_guide[0]
+                ):
+                    if min_onset is None or inactivity_bout[0] < min_onset:
+                        min_onset = inactivity_bout[0]
+                    if max_wakeup is None or inactivity_bout[1] > max_wakeup:
+                        max_wakeup = inactivity_bout[1]
+            if min_onset is not None and max_wakeup is not None:
+                sleep_windows.append(SleepWindow(onset=min_onset, wakeup=max_wakeup))
+
+        if not sleep_windows:
+            sleep_windows.append(SleepWindow(onset=[], wakeup=[]))
+
+        return sleep_windows
+
+    def _find_long_blocks(
+        self, below_threshold: np.ndarray, block_length: int = 360
+    ) -> np.ndarray:
+        """Helper function to find long blocks where SPT window is true.
+
+        This function uses the convolution of a kernel of 1s, of length block_length,
+        to find the continuous long blocks where SPT window is true. Where the
+        convolution value is == long_block_length that implies an overlap between the
+        kernel and the threshold signal of length long_block.
+
+        Args:
+            below_threshold: the 5-minute rolling median of the anglez difference
+                that is true when below the cutoff threshold.
+            block_length: the length of the long block that defines sleep, default is
+                30 minutes. (360 chunks of 5s)
+
+        Returns:
+            A numpy array with 1s indicating the identified SPT windows.
+        """
+        kernel = np.ones(block_length, dtype=int)
+        convolved = np.convolve(below_threshold, kernel, mode="same")
+        long_blocks_idx = np.where(convolved == block_length)[0]
+        sleep_idx_array = np.zeros(len(below_threshold))
+        sleep_idx_array[long_blocks_idx] = 1
+
+        return sleep_idx_array
+
+    def _fill_short_blocks(
+        self, sleep_idx_array: np.ndarray, gap_block: int = 720
+    ) -> np.ndarray:
+        """Helper function to fill gaps in SPT window that are less than 60 minutes.
+
+        We first find all the zeros in the sleep_idx_array, and then use np.diff to find
+        the gaps between those zeros, if np.diff !=0, that implies there is a gap
+        between 0s. We split slp_idx_array and find, and fill, the blocks that
+        are < gap_block.
+
+        Args:
+            sleep_idx_array: the array of SPT windows.
+            gap_block: the length of the gap that defines sleep, default is 60 minutes.
+                The units are chunks of 5s.
+
+        Returns:
+            A numpy array with 1s indicating the identified SPT windows.
+        """
+        zeros_idx = np.where(sleep_idx_array == 0)[0]
+        n_blocks = np.split(zeros_idx, np.where(np.diff(zeros_idx) != 1)[0] + 1)
```

Comment:
This is a list of blocks, not a number of blocks, right?

Reply:
Right, it's a list of arrays (break zeros_idx into a list of sub-arrays). Bad var name, will change